src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -844,13 +844,14 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
     // E0518: Inline attribute is allowed only on functions
     private fun checkInlineAttr(holder: RsAnnotationHolder, attr: RsAttr) {
         val metaItem = attr.metaItem
-        if (metaItem.name == "inline") {
-            val parent = attr.owner
 
-            if (parent !is RsFunction && parent !is RsLambdaExpr) {
-                RsDiagnostic.IncorrectlyPlacedInlineAttr(metaItem.path?.referenceNameElement ?: metaItem, attr)
-                    .addToHolder(holder)
-            }
+        if (metaItem.name != "inline") return
```

Comment:
Why do you not use `checkMetaItemName` here?

Reply:
For attributes like `inline`, using 
```rust
//- dep-proc-macro/lib.rs
    use proc_macro::TokenStream;

    #[proc_macro_attribute]
    pub fn inline(attr: TokenStream, item: TokenStream) -> TokenStream {
        item
    }
//- main.rs
    extern crate dep_proc_macro;

    use dep_proc_macro::inline;

    #[inline] // ambiguity
    type Foo = i32;
```
directly will actually lead to `E0659`.

For other explicit statements, 
```rust
//- main.rs
    extern crate dep_proc_macro;

    #[dep_proc_macro::inline] // metaItem.name == null
    type Foo = i32;
```
```rust
//- main.rs
    extern crate dep_proc_macro;

    use dep_proc_macro::inline as test;

    #[test] // metaItem.name == test
    type Foo = i32;
```

the plug-in currently does not resolve the name into `inline`, so only `metaItem.name != "inline"` is used here. I think it is enough.

Reply:
> directly will actually lead to E0659.

Yep, I'm aware about it. Moreover, https://github.com/intellij-rust/intellij-rust/pull/5582#discussion_r443247691 says about it directly (see code example). But it's exactly what we want - we shouldn't try to provide any analysis if there is `E0659` for the particular attribute. Yep, currently, we don't handle `E0659` but it will be handled eventually

> the plug-in currently does not resolve the name into inline

Yeah, our name resolution doesn't work properly in this case as well. But it doesn't make your changes worse, i.e. your current changes fix some rare false-positive errors and I think it's ok that we don't handle all similar false-positive errors because of the bug in name resolution. But we don't provide other false-positive errors so it's definitely an improvement